### PR TITLE
[#9] Notes Repository

### PR DESCRIPTION
**Pull request by lvyyln**

### CODE REVIEW

File: src/BetterRead.Shared.Repository/BookNotesRepository.cs [hunk 1]
```diff
@@ -1,18 +1,60 @@
...
+
+        public Note ConvertNote(IEnumerable<HtmlNode> note) =>
+            new Note()
+            {
+                 Id = Convert.ToInt32(note.FirstOrDefault()?.InnerText),
+                 Contents = note.Skip(1).Select(nt => nt.InnerText).ToList()
+            };
+    }
+
+    public static class Spliter
```
Comment (iTKerry) [inline src/BetterRead.Shared.Repository/BookNotesRepository.cs]: @lvyyln Please, rename this class to **EnumerableHelpers** and move it to _Common/Helpers_ (we have another project for such files)

File: src/BetterRead.Shared.Repository/BookNotesRepository.cs [hunk 2]
```diff
@@ -1,18 +1,60 @@
...
 namespace BetterRead.Shared.Repository
 {
     public class BookNotesRepository : IBookNotesRepository
     {
-        public Task<IEnumerable<Note>> GetNotesAsync(int bookId)
+        private readonly HtmlWeb _htmlWeb;
+
+        public BookNotesRepository()
+        {
+            _htmlWeb = new HtmlWeb {OverrideEncoding = Encoding.GetEncoding("windows-1251")};
```
Comment (iTKerry) [inline src/BetterRead.Shared.Repository/BookNotesRepository.cs]: @lvyyln I thought that we need apply encoding registry just like in another repositories, isn't it?

File: src/BetterRead.Shared.Repository/BookNotesRepository.cs [hunk 3]
```diff
@@ -1,18 +1,60 @@
...
+
+        public async Task<IEnumerable<Note>> GetNotesAsync(int bookId)
         {
             var url = string.Format(BookUrlPatterns.Notes, bookId);
+
+            var htmlDocument = await _htmlWeb.LoadFromWebAsync(url);
+            var documentNode = htmlDocument.DocumentNode.QuerySelectorAll("td.MsoNormal").FirstOrDefault()?.ChildNodes;
             
-            throw new System.NotImplementedException();
+            var temp = documentNode
```
Comment (iTKerry) [inline src/BetterRead.Shared.Repository/BookNotesRepository.cs]: @lvyyln please, never call your variables like **temp**

File: src/BetterRead.Shared.Repository/BookNotesRepository.cs [hunk 4]
```diff
@@ -1,18 +1,60 @@
...
+                .SplitOn(node => node.Name == "a")
+                .Select(g => g.Where(node => !(node is HtmlTextNode)))
+                .Where(g => g.Any());
+            var notes = temp.Select(note => ConvertNote(note));
```
Comment (iTKerry) [inline src/BetterRead.Shared.Repository/BookNotesRepository.cs]: @lvyyln I believe that you can merge this select with previous queries.

**Comment by iTKerry** [comment]
@lvyyln also there is some conflicts. can you resolve them?

File: src/BetterRead.Shared.Repository/BookNotesRepository.cs (hunk 2 above)
Comment (lvyyln) [reply src/BetterRead.Shared.Repository/BookNotesRepository.cs]: no, this is singleton service so it's one on whole application

**Comment by iTKerry** [comment]
@lvyyln any updates?

**Comment by lvyyln** [comment]
@iTKerry i resolved problems

**Comment by iTKerry** [comment]
@lvyyln **This branch has conflicts that must be resolved**, can you fix it?

**Pull request merged**